app/store/reducers/healthcareAuthoritiesReducer.ts

```diff
@@ -36,10 +38,28 @@ const healthcareAuthoritiesReducer = createReducer(initialState, (builder) =>
     })
     .addCase(
       getHealthcareAuthorities_success,
-      (state, { payload: { healthcareAuthorities } }) => {
+      (state, { payload: { healthcareAuthorities, usesCustomUrl } }) => {
         state.request.status = ApiStatus.SUCCESS;
         state.request.errorMessage = null;
-        state.availableAuthorities = healthcareAuthorities;
+        // This is a hack we use to hijack this reducer + action for
+        // using a testing url for the yaml instead of the environment one
+        if (usesCustomUrl) {
+          state.availableCustomAuthorities = healthcareAuthorities;
+        } else {
+          state.availableAuthorities = healthcareAuthorities;
+        }
+
+        // Ensure that we don't subscribe to non-existent HAs, if the yaml updates
+        const validIds = new Set();
+        state.availableAuthorities.forEach(({ internal_id }) => {
```

Comment:
```suggestion
        state.availableAuthorities.forEach(({ internalId }) => {
```
whats the reason for mixing cases here?

Reply:
This is part of the `HealthAuthority` data schema. The incoming data is all snake case, and the `internal_id` is set as part of the API layer, so from a consumer standpoint matches the payload. 
this is work revisiting for creating an API schema vs. app schema data validation / transformation step